gcapi/client.py

```diff
@@ -845,25 +847,72 @@ def update_archive_item(
             )
         )
 
-    def create_display_sets_from_images(
-        self, *, reader_study: str, display_sets: List[Dict[str, list]]
+    def _fetch_interface(self, slug):
+        try:
+            interface = yield from self.__org_api_meta.interfaces.detail(
+                slug=slug
+            )
+            return interface
+        except ObjectNotFound as e:
+            raise ValueError(
+                f"{slug} is not an existing interface. "
+                f"Please provide one from this list: "
+                f"https://grand-challenge.org/components/interfaces/reader-studies/"
+            ) from e
+        return interface
+
+    def _validate_display_set_values(self, values, interfaces):
+        invalid_file_paths = {}
+        for slug, value in values:
+            if interfaces.get(slug):
+                interface = interfaces[slug]
+            else:
+                interface = yield from self._fetch_interface(slug)
+                interfaces[slug] = interface
+            super_kind = interface["super_kind"].casefold()
+            if super_kind != "value":
+                if not isinstance(value, list):
+                    raise ValueError(
+                        f"Values for {slug} ({super_kind}) "
+                        "should be a list of file paths."
+                    )
+                if super_kind != "image" and len(value) > 1:
+                    raise ValueError(
+                        f"You can only upload one single file "
+                        f"to interface {slug} ({super_kind})."
+                    )
+                for file_path in value:
+                    if not Path(file_path).exists():
+                        invalid_file_paths[slug] = invalid_file_paths.get(
+                            slug, []
+                        )
+                        invalid_file_paths[slug].append(str(file_path))
+        if invalid_file_paths:
+            raise ValueError(f"Invalid file paths: {invalid_file_paths}")
+
+        return interfaces
+
+    def create_display_sets_from_values(
```

Comment:
This changes the public API so requires a major version bump and documentation. Is `create_display_sets_from_values` the best name for this? We had complaints that it wasn't obvious to use this to "create a display set" or "add/append stuff to a reader study".